lib/mix/lib/mix/tasks/format.ex

```diff
@@ -177,6 +177,10 @@ defmodule Mix.Tasks.Format do
     end
 
     {not_equivalent, not_formatted}
+  rescue
+    e ->
+      Mix.shell().error("mix format failed when parsing file: #{file}")
+      reraise e, System.stacktrace()
```

Comment:
We need to capture the stacktrace in the beginning of the rescue otherwise other errors can change it.

```elixir
stacktrace = System.stacktrace()
Mix.shell().error("mix format failed when parsing file: #{file}")
reraise e, stacktrace
```